arch.sh

```diff
@@ -6,43 +6,45 @@
 # originally written by Markus Schade
 # (c) 2013-2015, Hetzner Online GmbH
 #
-
+# changed and extended by Thore Bödecker, 2015-10-05
+# changed and extended by Tim Meusel
+#
 
 # setup_network_config "$device" "$HWADDR" "$IPADDR" "$BROADCAST" "$SUBNETMASK" "$GATEWAY" "$NETWORK" "$IP6ADDR" "$IP6PREFLEN" "$IP6GATEWAY"
 setup_network_config() {
-  if [ "$1" -a "$2" ]; then
+  if [ -n "$1" ] && [ -n "$2" ]; then
     # good we have a device and a MAC
     CONFIGFILE="$FOLD/hdd/etc/systemd/network/50-hetzner.network"
     UDEVFILE="$FOLD/hdd/etc/udev/rules.d/80-net-setup-link.rules"
 
-    echo -e "### Hetzner Online GmbH - installimage" > $UDEVFILE
-    echo -e "# device: $1" >> $UDEVFILE
-    echo -e "SUBSYSTEM==\"net\", ACTION==\"add\", DRIVERS==\"?*\", ATTR{address}==\"$2\", ATTR{dev_id}==\"0x0\", ATTR{type}==\"1\", KERNEL==\"eth*\", NAME=\"$1\"" >> $UDEVFILE
+    echo -e "### ${COMPANY} - installimage" > "$UDEVFILE"
+    echo -e "# device: $1" >> "$UDEVFILE"
+    printf 'SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", ATTR{address}=="%s", ATTR{dev_id}=="0x0", ATTR{type}=="1", KERNEL=="eth*", NAME="%s"\n' "$2" "$1" >> "$UDEVFILE"
 
-    echo -e "### Hetzner Online GmbH - installimage" > $CONFIGFILE
-    echo -e "# device: $1" >> $CONFIGFILE
-    echo -e "[Match]" >> $CONFIGFILE
-    echo -e "MACAddress=$2" >> $CONFIGFILE
-    echo -e "" >> $CONFIGFILE
+    echo -e "### $COMPANY - installimage" > "$CONFIGFILE"
+    echo -e "# device: $1" >> "$CONFIGFILE"
+    echo -e "[Match]" >> "$CONFIGFILE"
+    echo -e "MACAddress=$2" >> "$CONFIGFILE"
+    echo -e "" >> "$CONFIGFILE"
 
-    echo -e "[Network]" >> $CONFIGFILE
-    if [ "$8" -a "$9" -a "${10}" ]; then
+    echo -e "[Network]" >> "$CONFIGFILE"
+    if [ -n "$8" ] && [ -n "$9" ] && [ -n "${10}" ]; then
       debug "setting up ipv6 networking $8/$9 via ${10}"
-      echo -e "Address=$8/$9" >> $CONFIGFILE
-      echo -e "Gateway=${10}" >> $CONFIGFILE
-      echo -e "" >> $CONFIGFILE
+      echo -e "Address=$8/$9" >> "$CONFIGFILE"
+      echo -e "Gateway=${10}" >> "$CONFIGFILE"
+      echo -e "" >> "$CONFIGFILE"
     fi
 
-    if [ "$3" -a "$4" -a "$5" -a "$6" -a "$7" ]; then
+    if [ -n "$3" ] && [ -n "$4" ] && [ -n "$5" ] && [ -n "$6" ] && [ -n "$7" ]; then
```

Comment:
wouldn't at least checking if this are integers get some safety? like `declare -i 3 4 5 6 7`, maybe even bound checks if the numbers are between 0 and 255 or 8 and 32